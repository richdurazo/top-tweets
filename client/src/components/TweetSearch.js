import React, { useState, useEffect, useRef } from 'react';

const TweetSearch = props => {
    const { onGetTweets } = props;
    const [enteredFilter, setEnteredFilter] = useState('');
    const inputRef = useRef();

    useEffect(() => {
        const timer = setTimeout(() => {
            if (enteredFilter === inputRef.current.value) {
                const hasValue = enteredFilter.length && inputRef.current.value.length;
                let uri = hasValue ? `/list?q=${enteredFilter}` : `/list`;
                onGetTweets(enteredFilter, 'search', uri);
            }
        }, 500);
        return () => {
            clearTimeout(timer);
        }
    }, [enteredFilter, inputRef]);

    return (
        <div className="tweet-search__container">

            <input
                type="search"
                placeholder="Search by keyword"
                value={enteredFilter}
                ref={inputRef}
                onChange={event => setEnteredFilter(event.target.value)} />

        </div>
    );
}

export default React.memo(TweetSearch);
