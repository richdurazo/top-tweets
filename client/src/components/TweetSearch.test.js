import React from 'react';
import { shallow, mount } from 'enzyme';
import TweetSearch from './TweetSearch';

describe('TweetSearch component', () => {

    let onGetTweetsMock = jest.fn();

    beforeEach(() => {
        onGetTweetsMock.mockClear();
    });

    const container = shallow(<TweetSearch />);

    it('should match the snapshot', () => {
        expect(container.html()).toMatchSnapshot();
    });

    it('should have a search field', () => {
        expect(container.find('input[type="search"]').length).toEqual(1);
    });

    it('Should change value when onChange was called', () => {
        const wrapper = mount(<TweetSearch onGetTweets={onGetTweetsMock}/>);

        wrapper.find('input[type="search"]').simulate('change', {
          target: {
            value: 'I am searching for things',
          },
        });

        expect(wrapper.find('input[type="search"]').prop('value')).toEqual(
          'I am searching for things',
        );
        
    });

});