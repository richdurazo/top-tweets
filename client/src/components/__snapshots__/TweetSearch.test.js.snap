// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TweetSearch component should match the snapshot 1`] = `"<div class=\\"tweet-search__container\\"><input type=\\"search\\" placeholder=\\"Search by keyword\\" value=\\"\\"/></div>"`;
